### ISSUE TITLE
[Bug]: "No such table: pilot"

### ISSUE THREAD

**Issue by XanatosX** [XanatosX/IL2-Career-Reviver#49]
**Describe the bug**
If I run the command `.\IL2CareerToolset.exe save game pilot` after I connected the database I do get the following error `no such table: pilot`.

**To Reproduce**
1. Set the database for the tool
2. Run the command `.\IL2CareerToolset.exe save game pilot`
3. Get the error message `no such table: pilot`

**Expected behavior**
I do get a list of pilots shown in the console.

**Screenshots**
\-

**Desktop (please complete the following information):**
 - OS: Unknown

**Additional context**
The issue was found as part of the [forum thread][forum-thread]

[forum-thread]: https://forum.il2sturmovik.com/topic/82998-career-revive-tool-cli/?do=findComment&comment=1270379

**Comment by XanatosX** [comment]
The same issues did reappear for another use

https://forum.il2sturmovik.com/topic/82998-career-revive-tool-cli/#findComment-1331592

To investigate the issue I properly need a database which does throw the error.

**Comment by Dutchbook** [comment]
Here is a career file with the issue: https://drive.google.com/drive/folders/1TNZiIVmLskZr4kQ4ptMYmrdnjABy14Ce?usp=drive_link

**Comment by XanatosX** [comment]
First of all, thank you for providing a file for testing purpose.

Could you please change the link to a shared one without any requirement for an google account.

**Comment by XanatosX** [comment]
If you do not want to share the file like that you could use a tool like "https://wormhole.app/" to upload the database. Set the expire to something like 24 hours to make sure the data is getting removed after that period.

I did spend the time to update the project to the newest versions so I can continue working on it without any additional issues.

**Comment by XanatosX** [comment]
That is working, thank you very much, will check

**Comment by XanatosX** [comment]
There is a nightly build for that, could you check if it is working?

https://github.com/XanatosX/IL2-Career-Reviver/releases/tag/Develop-805

**Comment by Dutchbook** [comment]
The issue itself is solved and the table gets loaded, yet somehow I can't seem to revive the pilot through the app.

Some things I tried:

entering "IL2CareerToolset save game pilot --player" gives: "Error: Object reference not set to an instance of an object."
entering ÏL2CareerToolset save game pilot *database id* or *name* doesn't give an error code, but doesn't revive the pilot either.
Here is a screenshot of what I see: https://imgur.com/a/e7qAtYC

Am I using the wrong command?

**Comment by XanatosX** [comment]
That's interesting, I will take a look once again :)

Didn't use the tool that much since I made it public, sorry for the issues you are facing.

**Comment by XanatosX** [comment]
Seems like the pilot is not in a squadron, crashing the application.

**Comment by Dutchbook** [comment]
> That's interesting, I will take a look once again :)
> 
> Didn't use the tool that much since I made it public, sorry for the issues you are facing.

No problem! I managed to rescue my career through DBBrowser, but since you put the effort into making an app I thought it might as well get feedback to plug any holes.

**Comment by XanatosX** [comment]
Thank you very much, I'm a little bit confused as when I try to revive a pilot I do get two entries:

<img width="1614" height="516" alt="Image" src="https://github.com/user-attachments/assets/434a429b-e5d8-4c7d-81b2-052dd11d29bf" />

seems like it's listing one of the alive entries as well, hmm

**Comment by XanatosX** [comment]
Seems like I found the issue thanks to your dataset :)

<img width="1480" height="279" alt="Image" src="https://github.com/user-attachments/assets/2c081d33-a62e-4424-8022-c80a5b29b270" />

<img width="1533" height="693" alt="Image" src="https://github.com/user-attachments/assets/427bfd65-d9cc-44e1-b2af-732ba492f01b" />

<img width="1479" height="273" alt="Image" src="https://github.com/user-attachments/assets/d2d880df-42a5-4545-a53f-f4a9b6b28982" />

Somehow the data structure is weird for that pilot as it cannot find the airfield because the squadron number is not matching. Did you use any external tool to create the pilot?

**Comment by XanatosX** [comment]
Changes are implemented, waiting for the nightly build to get triggered in a few hours. Sadly you cannot test it anymore but glad you solved it by yourself.

Hopefully someone can test the newest release and confirm it working.

**Comment by Dutchbook** [comment]
> 
> Somehow the data structure is weird for that pilot as it cannot find the airfield because the squadron number is not matching. Did you use any external tool to create the pilot?

Nope, none. But my career tells me I'm I./JG 3 out of Pitomnik, yet the rest of the campaign the game used Morozovskaya North as my home base with a transfer to Morozovskaya South later on. I thought the game intended to do this, since Pitomnik was historically under constant bombardment at this time. Must be a bug in the game?

Don't worry, i'll still test out the build :)

**Comment by XanatosX** [comment]
Build was created
 https://github.com/XanatosX/IL2-Career-Reviver/releases/tag/Develop-807

**Comment by Dutchbook** [comment]
Should I enter an argument behind the IL2CareerToolset.exe save game pilot [pilot name or database id]? It still doesn't seem to be working with the latest build...

**Comment by XanatosX** [comment]
So first of all if you use the tool for the first time you need to set the database with

`settings manuel "path to il2"`

This will set the database to work on. you can also use `settings auto` but that will scan your whole pc for that folder, you need to decide if you really want that.

After you did set the database you should be able to see your pilots via

`save game pilot -p`

Where at least one will be marked as dead as you want to revive it, note down the name.

run 

`save game revive`

You will get a warning that this could corrupt the database and you should create a backup first. After that it should list the dead pilot, if selected it will ask if you really want to revive him, if confirmed the process will start.

Hope that does help you.

**Comment by Dutchbook** [comment]
Ah, that explains it! The "save game revive" command isn't in the Readme or the "-h" of the program.

Now i'm getting the following error, even after a reboot:
Please close the game to unlock the database before you continue!

== What will happen ==

This application will revive the pilot and reset the date back to the last date he was alive.
It will clear the sortie and mission where the pilot died. The Game will create a new mission for your pilot,
this new mission is most likely different to the one you did die in.
Do you accept this warning? [y/n] (y): y
Do you really want to revive [NAME] [y/n] (y): y
Create Database backup
Error: The process cannot access the file 'C:\Program Files\IL-2 Sturmovik Great Battles\data\Career\cp.db' because it
is being used by another process.

But I'll believe it though, if it was up to me I would just make the latest nightly build the new finished version and call it a day. If I accidentally mess up my save again, I'll install it on my laptop and try it there and report if it doesn't work :) 

Thanks for the effort!

**Comment by XanatosX** [comment]
Okay to be honest I do not know why it thinks the database is locked by another process. Might be an issue with the backup down beforehand.

But this only copies the database so it should be unlocked as the next process is starting. But I think you are right I will make this the new version and hopefully the error is only something you got. If not someone will possible create an issue or complain in the forum.

Thank you very much for taking you time to test the versions and the tool itself, even after it did nothing for you.

I will close this issue for now.
